torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -102,8 +100,29 @@ TensorPipeAgent::~TensorPipeAgent() {
 }
 
 void TensorPipeAgent::startImpl() {
-  context_->registerTransport(
-      1, "tcp", std::make_shared<tensorpipe::transport::uv::Context>());
+  auto uvContext = std::make_shared<tensorpipe::transport::uv::Context>();
+
+  tensorpipe::Error error;
```

Comment:
Would it make sense to move the below logic to a separate helper function that returns uvAddress (for better modularity)?